### PR TITLE
feat(spotlight): adding in a deeplinks for spotlight

### PR DESCRIPTION
**Pull request by bassrock**
## Summary
* Enable deep linking from core spotlight search results to things in your list. Indexing is done by https://github.com/Pocket/pocket-ios/blob/6d0d6dce14a584171d794c4c44fd5f286a82c82d/PocketKit/Sources/Sync/CoreDataSpotlightDelegate.swift#L10 with CoreData automatically.

## References 
* https://mozilla-hub.atlassian.net/browse/POCKET-9379

## Implementation Details
* I went short/sweet and a bit hacky. Coredata continues the activity with a CoreData URI, so i turn that into an object, grab the url, and then let it run through our router to take a user to the item. I am open to better suggestions here, but this seemed like the easiest path for now.

## Test Steps
* Log into your Pocket account, as items load in, CoreData will auto spotlight index them in out SpotlightCoreDataDelegate.
* Search in spotlight for something in your list.
* Tap it, and you should be taken to the url within your list.

## PR Checklist:
- [ ] Added Unit / UI tests
- [x] Self Review (review, clean up, documentation, run tests)
- [x] Basic Self QA

### CODE REVIEW

File: PocketKit/Sources/PocketKit/Main/RootView.swift [hunk 1]
```diff
@@ -66,12 +67,20 @@ public struct RootView: View {
                 .onOpenURL { url in
                     model.handle(url)
                 }
-                .onContinueUserActivity("NSUserActivityTypeBrowsingWeb", perform: { userActivity in
+                .onContinueUserActivity(NSUserActivityTypeBrowsingWeb, perform: { userActivity in
```
Comment (bassrock) [inline PocketKit/Sources/PocketKit/Main/RootView.swift:70]: Switched this to Apple's hard to find constant.

File: PocketKit/Sources/PocketKit/Home/HomeViewModel.swift [hunk 2]
```diff
@@ -433,6 +434,9 @@ extension HomeViewModel {
                 recommendationId: recommendationId,
                 destination: destination
             ))
+        case .spotlight:
+            // Spot light never indexes recs.
+            Log.breadcrumb(category: "spotlight", level: .warning, message: "Somehow entered slate open from Spotlight, which should not happen")
```
Comment (nzeltzer) [inline PocketKit/Sources/PocketKit/Home/HomeViewModel.swift:439]: 💙

**Review by nzeltzer — APPROVED** [review]
lgtm

**Review by Gio2018 — APPROVED** [review]
Thanks for refactoring `RootView` @bassrock ! Looks great to me!

**Pull request merged**